### ISSUE TITLE
False positive TCH002 check

### ISSUE THREAD

**Issue by Olegt0rr** [charliermarsh/ruff#2870]
Since I import `cast` and use it in the code - `TCH002` rule stop working...

## Wrong behaviour (with casting)

```python
from typing import cast

from aiogram.types import Message, User


async def handle_start(message: Message) -> None:
    """Handle /start command."""
    user = cast(User, message.from_user)
    await message.answer(f"Hi, {user.full_name}")

```

There's no `TCH002` error about `Message`

## Right behaviour (without casting)

```python
from aiogram.types import Message


async def handle_start(message: Message) -> None:
    """Handle /start command."""
    user = message.from_user
    await message.answer(f"Hi, {user.full_name}")

```

TCH002 Move third-party import `aiogram.types.Message` into a type-checking block

**Comment by spaceone** [comment]
I think that is correct:
```python
from typing import TYPE_CHECKING, cast


if TYPE_CHECKING:
        from aiogram.types import Message, User


def handle_start(message: Message) -> None:
    """Handle /start command."""
    user = cast(User, message.from_user)
    message.answer(f"Hi, {user.full_name}")


handle_start(type('foo', (object,), {'answer:': lambda m: print(m)}))
```

```
$ python3 foo.py
Traceback (most recent call last):
  File "foo.py", line 8, in <module>
    def handle_start(message: Message) -> None:
NameError: name 'Message' is not defined
```

**Comment by Olegt0rr** [comment]
You forgot to quote it...

try this:

```python
from typing import TYPE_CHECKING, cast
from aiogram.types import User

if TYPE_CHECKING:
        from aiogram.types import Message


def handle_start(message: "Message") -> None:
    """Handle /start command."""
    user = cast(User, message.from_user)
    message.answer(f"Hi, {user.full_name}")


handle_start(type('foo', (object,), {'answer:': lambda m: print(m)}))
```

**Comment by spaceone** [comment]
alright, only `Message` should be in the `TYPE_CHECKING` block.

**Comment by Olegt0rr** [comment]
So in my first message in `Wrong behaviour` section there's no TCH002 error

**Comment by charliermarsh** [comment]
@Olegt0rr - This is because we have "strict mode" disabled by default. So if you import multiple members from a module, and at least one of them is required at runtime, we don't flag the others.

You can change this behavior like:

```toml
[tool.ruff.flake8-type-checking]
strict = true
```

Which gives you:

```
foo.py:3:27: TCH002 Move third-party import `aiogram.types.Message` into a type-checking block
```

**Comment by Olegt0rr** [comment]
@charliermarsh, thanks for the explanation!

**Comment by charliermarsh** [comment]
@Olegt0rr - No prob! Let me know if you run into issues with it. The reasoning, IIRC, is that if you're _already_ importing `aiogram.types` to get `User`, it's no more expensive to import `Message` too at runtime, so it's often not considered worthwhile to enforce.
